### PR TITLE
First Implementation of a better filter for the food page

### PR DESCRIPTION
**Pull request by FabianSimonGross**
First Implemenation of the Issue #110

### CODE REVIEW

**Review by Robert27 — DISMISSED** [review]

File: rogue-thi-app/pages/food.js [hunk 1]
```diff
@@ -104,222 +125,323 @@ export default function Mensa () {
...
+            <Modal.Body>
+              <h5>Anmerkungen</h5>
+              {showMealDetails?.flags === null && 'Unbekannt.'}
+              {showMealDetails?.flags?.length === 0 && 'Keine.'}
+              <ul>
+                {showMealDetails?.flags?.map(flag => (
+                    <li key={flag}>
+                      <strong>{flag}</strong>
+                      {' – '}
+                      {flagMap[flag] || FALLBACK_ALLERGEN}
```
Comment (M4GNV5) [inline rogue-thi-app/pages/food.js:224]: Here we could insert the handshake icon as well as green color, when the flag is one of the preferred ones. Similar to how the allergenes turn red (see line 234-238)

File: rogue-thi-app/pages/food.js [hunk 2]
```diff
@@ -104,222 +125,323 @@ export default function Mensa () {
...
+              {showMealDetails?.allergens === null && 'Unbekannt.'}
+              {showMealDetails?.allergens?.length === 0 && 'Keine.'}
+              <ul>
+                {showMealDetails?.allergens?.map(key => (
+                    <li key={key} style={{ color: containsSelectedAllergen([key]) && COLOR_WARN }}>
+                      {containsSelectedAllergen([key]) && (
+                          <span>
                       <FontAwesomeIcon icon={faExclamationTriangle} color={COLOR_WARN} />
-                      {' '}
+                            {' '}
```
Comment (M4GNV5) [inline rogue-thi-app/pages/food.js:238]: The indentation here is wrong.

File: rogue-thi-app/pages/food.js [hunk 3]
```diff
@@ -104,222 +125,323 @@ export default function Mensa () {
...
+                      <FontAwesomeIcon title="Allergene" icon={faAllergies} fixedWidth />
+                    </Button>
+                  </div>
+              </Modal.Header>
+
+              <Modal.Body>
+                <span>
+                  {Object.entries(allergenMap).map(([mapkey, mapvalue]) => (
+                    Object.entries(allergenSelection).map(([key, value]) => (
+                          <span key={key} >{mapkey === key && value === true && mapkey + ' - ' + mapvalue + ', '}</span>
```
Comment (M4GNV5) [inline rogue-thi-app/pages/food.js:356]: Here you create <span> elements for all allergenes, even the ones not selected.
I do not think we need those elements at all, a simple comma separated string should be enough.

Something like this:
```javascript
Object.entries(allergenSelection).filter(x => x[1]).map(x => allergenMap[x[0]]).join(', ')
```

File: rogue-thi-app/pages/food.js [hunk 4]
```diff
@@ -104,222 +125,323 @@ export default function Mensa () {
...
+              </Modal.Header>
+
+              <Modal.Body>
+                <span>
+                  {Object.entries(flagMap).map(([mapkey, mapvalue]) => (
+                    Object.entries(preferencesSelection).map(([key, value]) => (
+                          <span key={key} >{mapkey === key && value === true && mapvalue + ', '}</span>
+                    ))
+                  ))
+                  }
```
Comment (M4GNV5) [inline rogue-thi-app/pages/food.js:381]: See above comment regarding allergeneMap and allergeneSelection

**Review by M4GNV5 — CHANGES_REQUESTED** [review]

File: rogue-thi-app/pages/food.js (hunk 2 above)
Comment (FabianSimonGross) [reply rogue-thi-app/pages/food.js:238]: Will be fixed in next push.

File: rogue-thi-app/pages/food.js (hunk 3 above)
Comment (FabianSimonGross) [reply rogue-thi-app/pages/food.js:356]: Will be fixed in next push.

File: rogue-thi-app/pages/food.js (hunk 4 above)
Comment (FabianSimonGross) [reply rogue-thi-app/pages/food.js:381]: Will be fixed in next push.

File: rogue-thi-app/pages/food.js (hunk 1 above)
Comment (FabianSimonGross) [reply rogue-thi-app/pages/food.js:224]: Also coming with the next push

**Comment by FabianSimonGross** [comment]
I currently have some trouble with configuring my IDE to meet the requirements for indentation.
Maybe you look again. Until then I'll try getting my IDE on the requirements.

**Review by M4GNV5 — APPROVED** [review]

**Pull request merged**